cgen: `match` a rune inside an `if/else` stmt used as the value of a var

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
```
OS: macos, macOS, 12.3.1, 21E258
Processor: 8 cpus, 64bit, little endian, Apple M1
CC version: Apple clang version 13.1.6 (clang-1316.0.21.2)

getwd: /Users/hunam/Documents/dev/go2v
vmodules: /Users/hunam/.vmodules
vroot: /Applications/v
vexe: /Applications/v/v
vexe mtime: 2022-04-19 10:49:51
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 1fc54a1.a318a2e

Git version: git version 2.32.0 (Apple Git-132)
Git vroot status: weekly.2022.16-6-ga318a2e0 (3 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-unknown-unknown de82a130
```

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
**What did you do?**
```v
a := if true {
	match `a` {
		`a` { 0 }
		else { 1 }
	}
} else {
	3
}
```

**What did you expect to see?**
no error
 
**What did you see instead?**
```v
==================
   ^~~~~~~~~~~~~~~
/tmp/v_501/a.7448932374503026173.tmp.c:11358:26: error: expected ')'
        int a = (true ? (               rune _t1 = 'a';
                                             ^
/tmp/v_501/a.7448932374503026173.tmp.c:11358:18: note: to match this '('
        int a = (true ? (               rune _t1 = 'a';
                        ^
/tmp/v_501/a.7448932374503026173.tmp.c:11358:35: error: expected expression
        int a = (true ? (               rune _t1 = 'a';
                                                      ^
/tmp/v_501/a.7448932374503026173.tmp.c:11358:35: error: expected ':'
        int a = (true ? (               rune _t1 = 'a';
...
==================
```

--- a/vlib/v/gen/c/if.v
+++ b/vlib/v/gen/c/if.v
@@ -20,6 +20,9 @@ fn (mut g Gen) need_tmp_var_in_if(node ast.IfExpr) bool {
 					if is_noreturn_callexpr(stmt.expr) {
 						return true
 					}
+					if stmt.expr is ast.MatchExpr {
+						return true
+					}
 					if stmt.expr is ast.CallExpr {
 						if stmt.expr.is_method {
 							left_sym := g.table.sym(stmt.expr.receiver_type)

--- a/vlib/v/tests/if_expr_with_nested_match_expr_test.v
+++ b/vlib/v/tests/if_expr_with_nested_match_expr_test.v
@@ -1,0 +1,12 @@
+fn test_if_expr_with_nested_match_expr() {
+	a := if true {
+		match `a` {
+			`a` { 0 }
+			else { 1 }
+		}
+	} else {
+		3
+	}
+	println(a)
+	assert a == 0
+}

